Frontend/src/app/maps/bing-maps.service.ts

```diff
@@ -44,266 +50,393 @@ export class BingMapsService {
+        }
+
+        const script = document.createElement('script');
+        script.type = 'text/javascript';
+        script.async = true;
+        script.defer = true;
+
+        const mapsCallback = 'bingMapsCallback';
+        script.src = `https://www.bing.com/api/maps/mapcontrol?branch=release&clientApi=bingmapsfleettracker&callback=${mapsCallback}`;
+
+        this.loadPromise = new Promise<
+            void
+            >((resolve: Function, reject: Function) => {
+                window[mapsCallback] = () => {
+                    resolve();
+                };
+                script.onerror = (error: Event) => {
+                    console.error('maps script error' + error);
+                    reject(error);
+                };
+            });
+
+        document.body.appendChild(script);
+
+        return this.loadPromise;
+    }
+
+    private centerMapOnMedian(points: Point[]) {
+        if (points.length) {
+            points.sort(function (x, y) {
+                return x.longitude - y.longitude;
+            });
+            const mid = Math.floor(points.length / 2);
+            this.map.setView({
+                center: new Microsoft.Maps.Location(
+                    points[mid].latitude,
+                    points[mid].longitude
+                )
+            });
+        }
+    }
+
+    private centerMapOnMean(points: Point[]) {
+        if (points.length) {
+            let centerLat = 0;
+            let centerLong = 0;
+            points.forEach(function (p) {
+                centerLat += p.latitude;
+                centerLong += p.longitude;
+            });
 
-            const locations = new Array<Microsoft.Maps.Location>();
+            centerLat /= points.length;
+            centerLong /= points.length;
 
-            // tslint:disable-next-line:forin
-            for (const g in geofences) {
-                const geofence = geofences[g];
+            this.map.setView({
+                center: new Microsoft.Maps.Location(
+                    centerLat,
+                    centerLong
+                )
+            });
+        }
+    }
 
-                locations.splice(0, locations.length);
-                locations.push.apply(locations,
-                    this.drawGeofencePolygon(geofence, 'rgba(86,182,145,0.4)', BingMapsService.colors[0]));
+    private timeConverter(timestamp) {
+        const a = new Date(timestamp);
+        const months = [
+            'Jan',
+            'Feb',
+            'Mar',
+            'Apr',
+            'May',
+            'Jun',
+            'Jul',
+            'Aug',
+            'Sep',
+            'Oct',
+            'Nov',
+            'Dec'
+        ];
+        const year = a.getFullYear();
+        const month = months[a.getMonth()];
+        const day = a.getDate();
+        const hour = a.getHours();
+        const min = a.getMinutes();
+        const sec = a.getSeconds();
+        return `${day} ${month} ${hour}:${min}:${sec}`;
```

Comment:
use
`a.toLocaleString('en-US', { year: 'numeric', month: 'short', day: 'numeric', hour: 'numeric', minute: 'numeric', second: 'numeric' });`

For more details: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Date/toLocaleString